app/src/main/java/io/novafoundation/nova/app/root/navigation/Navigator.kt

```diff
@@ -350,9 +351,15 @@ class Navigator(
         navController?.navigate(R.id.action_mainFragment_to_swapFlow, AssetSwapFlowFragment.getBundle(payload))
     }
 
-    override fun openSwapSettings(assetPayload: AssetPayload) {
-        val payload = SwapSettingsPayload(assetPayload)
-        navController?.navigate(R.id.action_swapFlowFragment_to_swapSettings, SwapMainSettingsFragment.getBundle(payload))
+    override fun openSwapSetupAmount(assetPayload: AssetPayload) {
+        val returnTo = when (navController?.currentDestination?.id) {
+            R.id.swapFlowFragment -> SwapFinishFlowDestination.BALANCE_LIST
+            R.id.balanceDetailFragment -> SwapFinishFlowDestination.BALANCE_DETAILS
```

Comment:
Hm I thought we wanted to always to open balance details regardless of the origin

Reply:
Fixed